zone/zonedb.cpp

```diff
@@ -214,7 +214,7 @@ void ZoneDatabase::LoadWorldContainer(uint32 parent_id, EQ::ItemInstance* contai
 	}
 
 	const auto& l = ObjectContentsRepository::GetWhere(
-		*this,
+		database,
```

Comment:
This is a bizarre pattern. Database functions internally should always use `*this` and the caller should specify which database class instance to use e.g., `database.LoadWorldContainer(...)` or `content_db.LoadWorldContainer(...)`. It looks like these are already using the state `database` in callers though, is there something else going on?

Reply:
Not 100% sure what is going on there. I just know it was calling the content database.

Reply:
Object::PutItem
  calls database.SaveWorldContainer (ok)
  calls Save()
    Object::Save calls content_db.UpdateObject() or content_db.AddObject()
      ZoneDatabase::UpdateObject() and ZoneDatabase::AddObject() both call SaveWorldContainer() on their own content_db instance
      SaveWorldContainer() calls DeleteWorldContainer() on that same content_db instance

Probably the calls to SaveWorldContainer() in UpdateObject/AddObject could be changed to explicitly use
database.SaveWorldContainer() but I don't know what exactly is being saved here.

It looks like old code also did this so it's not immediately clear why this would start warning so often now unless Object::Save is now being called from somewhere it didn't used to be?

Reply:
Nevermind I see, the old code called `database.QueryDatabase(query)` explicitly in `SaveWorldContainer` and `DeleteWorldContainer`. I think this one should be reverted and the same thing done.

Reply:
Care to put in a pull request to adjust that?